src/components/routes/ViewConfig.tsx

```diff
@@ -24,19 +25,37 @@ interface DispatchProps {
     setTree: (components: ComponentConfig[]) => void;
 }
 
-type FullProps = MetaDataProps & StateProps & DispatchProps & RouteComponentProps<any> & FormWrapperProps;
+type FullProps = MetaDataProps & StateProps & DispatchProps & RouteComponentProps<any>;
+
+type GraphProps = ChildProps<FullProps, { dataSet: string; collectionUrl: string; viewConfig: ComponentConfig[] }>;
 
 const Section = styled.div`
     width: 100%;
     padding-bottom: 3rem;
 `;
 
-const exampleData: ComponentConfig[] = [];
-
-const ViewConfig: SFC<FullProps> = props => {
+const ViewConfig: SFC<GraphProps> = props => {
     const onSubmit = () => {
-        const tree = props.denormalizeTree();
-        console.log(tree);
+        if (
+            !props.mutate ||
+            !props.metadata ||
+            !props.metadata.dataSetMetadata ||
```

Comment:
Can you put a todo here that this entire check should be replaced with a NotFound HoC?